code/obj/machinery/computer/robotics.dm

```diff
@@ -16,48 +16,12 @@
 
 
 /obj/machinery/computer/robotics/attackby(obj/item/I as obj, user as mob)
-	if (isscrewingtool(I))
-		if (perma)
-			boutput(user, "<span class='alert'>The screws are all weird safety-bit types! You can't turn them!</span>")
-			return
-		playsound(src.loc, "sound/items/Screwdriver.ogg", 50, 1)
-		if(do_after(user, 2 SECONDS))
-			if (src.status & BROKEN)
-				boutput(user, "<span class='notice'>The broken glass falls out.</span>")
-				var/obj/computerframe/A = new /obj/computerframe( src.loc )
-				if(src.material) A.setMaterial(src.material)
-				var/obj/item/raw_material/shard/glass/G = unpool(/obj/item/raw_material/shard/glass)
-				G.set_loc(src.loc)
-				var/obj/item/circuitboard/robotics/M = new /obj/item/circuitboard/robotics( A )
-				for (var/obj/C in src)
-					C.set_loc(src.loc)
-				M.id = src.id
-				A.circuit = M
-				A.state = 3
-				A.icon_state = "3"
-				A.anchored = 1
-				qdel(src)
-			else
-				boutput(user, "<span class='notice'>You disconnect the monitor.</span>")
-				var/obj/computerframe/A = new /obj/computerframe( src.loc )
-				if(src.material) A.setMaterial(src.material)
-				var/obj/item/circuitboard/robotics/M = new /obj/item/circuitboard/robotics( A )
-				for (var/obj/C in src)
-					C.set_loc(src.loc)
-				M.id = src.id
-				A.circuit = M
-				A.state = 4
-				A.icon_state = "4"
-				A.anchored = 1
-				qdel(src)
-
-	else
-		src.attack_hand(user)
+	if (perma && isscrewingtool(I))
+		boutput(user, "<span class='alert'>The screws are all weird safety-bit types! You can't turn them!</span>")
+		return
```

Comment:
I realize this is probably an existing bug, but does this mean that I can't fix the glass of a broken perma robotics control?

Reply:
Correct, is that a bug? It may make more sense to fix that with a different action than other computers use. The step to replace the glass turns the computer into a frame which would allow it to be fully deconstructed. Perhaps I can set it to an action for when hitting the computer with a glass sheet? That would prevent it from being deconstructed but allow glass to be replaced. A would imagine a more thorough solution may fit better in its own PR. If you disagree, I can work on including something here.

Reply:
Its own PR is fine. This is an existing bug.